lib/HttpKernel/CacheWarmer/PimcoreCoreCacheWarmer.php

```diff
@@ -79,7 +79,8 @@ private function getClassesFromDirectory(string $dir, string $excludePattern, st
                 $className = preg_replace('@\.php$@', '', $className);
                 $className = str_replace(DIRECTORY_SEPARATOR, '\\', $className);
 
-                if (class_exists($className)) {
+                // do not load all classes here / causes memory issues
+                if (class_exists($className, false)) {
```

Comment:
At this time, no of the classes will be loaded already, so this change makes the whole cache-warmer obsolete.

Reply:
The cache warmer is only returning an array of classes, why would this change make the cache warmer obsolete?

The real problem is, that the memory consumption is causing cache:clear to fail because we are loading the whole lib, models and classes. 
This would also cause problems in instances with many classes.

Reply:
@mattamon is right, these are classes that should be *preloaded* by PHP:

https://symfony.com/doc/current/reference/dic_tags.html#kernel-cache-warmer

Reply:
https://symfony.com/doc/current/reference/dic_tags.html#kernel-cache-warmer
Also according to symfony, you do not need to load the classes inside here, just return the array, what should be considered.

Reply:
That's true, but with your change, no class is returned anymore, because `class_exists($className, false)` is always `false` because it disables autoloading (and non of the classes is loaded at this time), therefore no classes will get added to the return value, ergo -> obsolete.

Reply:
```php
    if (class_exists($className, false)) {
           p_r("In here");
           $classes[] = $className;
}
```
I just tested it with autoload false and the debug statement.
![image](https://github.com/pimcore/pimcore/assets/38959016/4c48efb7-9dfa-46a5-8c18-b8d6013cbfa0)

At least for my system it still works

Reply:
@brusch, @cipribucur and I tested it more extensively and discovered some more problems with `class_exists`. 
`class_exists`  excludes traits and interfaces. We changed now to a different approach and do not use the `class_exists` altogether since it is covered in the preloader anyways. https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Dumper/Preloader.php#L80